deploy/runtools/run_farm.py

```diff
@@ -111,30 +114,96 @@ class RunFarm(metaclass=abc.ABCMeta):
 
     Attributes:
         args: Set of options from the 'args' section of the YAML associated with the run farm.
+        default_simulation_dir: default location of the simulation dir on the run farm host
```

Comment:
I personally don't think I like moving all this logic up into the `RunFarm` base class. Future run farms could have a different implementation on how to support metasims, different ways to give a 'spec', non-standard default simulation dirs, etc. Additionally, having the base class be as simple as possible makes it easier to understand the code (IMO). I think it would be nice to move this code into a separate base class such that the inheritance structure is `RunFarm -> RunFarmWithSpecs -> {EC2, ExProv}`.

Reply:
+1 to @davidbiancolin , leaving this as is